### ISSUE TITLE
Sync and merge contractor journals across (renter-folder) devices

### ISSUE THREAD

**Issue by MeijeSibbel** [GooBox/goobox-sync-sia#8]
If a user runs two Goobox instances with a full-SIA-node as the back-end on two different machines using the same seed, only the node where the file is uploaded will be able to access the file because the other node doesn't have the updated contractor journals, so without the contractor journals the files can't be downloaded. Even though seed only file recovery is on the 1-2 year roadmap, there are a few other options we could explore.

1. Create a central service that acts as a bridge between the different nodes, so every time a file is changed on one node, its uploads the renter-directory to the server, which is then downloaded on the other node. 

2. Look into this: https://trello.com/c/J91JmVaO/73-ability-to-recover-files-from-an-outdated-backup-of-the-renter-directory, this might allow for automatic rebuilding of a incomplete renter-directory. Maybe this will require us to sync the renter-directory only once on the other node. Please follow the discussion in the #core-dev channel on the SIA discord. 

3. Use https://trello.com/c/7litw6xv/4-simple-file-sharing-share-files-with-other-sia-users to share the renter directory with the second node. The thing here is how are we going to implement this in a way the other node sees the file sharing link.

Please give me your comments on this @kaloyan-raev and @jkawamoto . As the consumer edition will be hosted nodes, it might be a good idea of thinking of the best way to implement this that we don't have to build a entire new system later. We have to find a balance between what's easiest to implement and what's most secure.

**Comment by kaloyan-raev** [comment]
I believe we should look at this from the perspective of Goobox goals. So, what do we want to achieve with Goobox? As far as I get it, we want users to have their data easily accessible on all their devices, much like with Dropbox. And "all user devices" is pretty broad: desktop workstations, laptops, tablets, phones, ebook readers, smart TVs, etc.

How many of these devices can host a full SIA node? Downloading tens of GBs of a blockchain, maintaining contracts, etc. Even on a desktop workstation or laptop, it is much of a pain to run a SIA node. Obviously, if we want Goobox to offer easy-to-use SIA storage, we should host the SIA node for the user. Or use a 3rd party service like SiaCDN to do it for us.

So, if we go in the direction of hosting the SIA node for the user then I would expect that a single highly available SIA node would be enough. Then we don't need to handle issue like synchronizing contractor journals or whatever between multiple nodes. We just need one node for the user and make it available to all her devices, anywhere in the world.

I recommend that we first take a look at how SiaCDN would fit in the picture. Perhaps, it already has all capabilities we need to implement Goobox. Maybe not.

**Comment by MeijeSibbel** [comment]
I agree, sia mentioned that they will implement seed-only file recovery in Q1 of 2018, this means that it will for the community-gui at least be easier for community members to run the same node on different machines by entering the seed into each machine at start, this is some logic we have to then add to the community-GUI to accept a seed for sia the same way we accept login details for Storj. For the consumer version hosted nodes is indeed the only option so this is not relevant to that system. I will close this issue for now and once seed-based recovery is implemented i will create a new issue.
